src/golang/lib/notification/notification.go

```diff
@@ -55,8 +100,32 @@ func NewNotificationFromIntegration(
 			return nil, err
 		}
 
-		return newSlackNotification(slackConf), nil
+		return newSlackNotification(integrationObject, slackConf), nil
 	}
 
 	return nil, ErrIntegrationTypeIsNotNotification
 }
+
+// `ShouldSend` determines if a notification at 'level' passes configuration
+// specified by `thresholdLevel`.
+// 'info' and 'neutral' will get through regardless of threshold.
+// And 'info' or 'neutral' threshold lets everything through.
+// Other states will follow the severity ordering.
+func ShouldSend(
+	thresholdLevel shared.NotificationLevel,
+	level shared.NotificationLevel,
+) bool {
```

Comment:
I have a slight preference of making this lowercase and only use it within send() to simplify the callers. The Send() method noops anyways if content == nil, so it could follow the similar pattern if the severity isn't high enough.

Reply:
The issue is whether to send is decided by caller due to the 'default - override' behavior (see the `sendNotification` implementation in `engine` package). An [alternative design](https://github.com/aqueducthq/aqueduct/pull/923) is to allow caller to pass in the overrides so that we can do this `shouldSend` within the package. But I found the API interface in this alternative is also confusing as we need to explain what 'override' means:

```
wfSettings := dag.Settings() // overrides
notification.Send(
  msg,
  level, // notification level
  wfSettings.Level, // override value
)
```